daemon/containerd/image_exporter.go

```diff
@@ -20,18 +27,64 @@ import (
 //
 // TODO(thaJeztah): produce JSON stream progress response and image events; see https://github.com/moby/moby/issues/43910
 func (i *ImageService) ExportImage(ctx context.Context, names []string, outStream io.Writer) error {
+	platform := platforms.AllPlatformsWithPreference(cplatforms.Default())
 	opts := []archive.ExportOpt{
-		archive.WithPlatform(platforms.Ordered(platforms.DefaultSpec())),
 		archive.WithSkipNonDistributableBlobs(),
+
+		// This makes the exported archive also include `manifest.json`
+		// when the image is a manifest list. It is needed for backwards
+		// compatibility with Docker image format.
+		// The containerd will choose only one manifest for the `manifest.json`.
```

Comment:
No, the main consumer of the `manifest.json` is of course the "graphdriver Docker", but this one is actually about the containerd choosing only one manifest to include in the `manifest.json`:
https://github.com/containerd/containerd/blob/1c90a442489720eec95342e1789ee8a5e1b9536f/images/archive/exporter.go#L232-L234

It sorts all manifests by their platform, according to the platform matcher `Less` and chooses the "least" one.
In this case we prefer it to choose the default (host) platform, but if that one won't be exported, then let it choose whatever because for now we don't have any CLI flags for the user to communicate what is his intent.